gui/codeeditor.cpp

```diff
@@ -133,41 +167,58 @@ void Highlighter::highlightBlock(const QString &text)
     }
 }
 
-
-CodeEditor::CodeEditor(QWidget *parent,
-                       CodeEditorStyle *widgetStyle /*= nullptr*/) :
-    QPlainTextEdit(parent)
+void Highlighter::applyFormat( HighlightingRule &rule )
 {
-    if (widgetStyle) mWidgetStyle = widgetStyle;
-    else mWidgetStyle = new CodeEditorStyle(defaultStyle);
+    switch( rule.ruleRole )
+    {
+        case RuleRole::Keyword:
+            rule.format = mKeywordFormat;
+            break;
+        case RuleRole::Class:
+            rule.format = mClassFormat;
+            break;
+        case RuleRole::Comment:
+            rule.format = mSingleLineCommentFormat;
+            break;
+        case RuleRole::Quote:
+            rule.format = mQuotationFormat;
+            break;
+        case RuleRole::Symbol:
+            rule.format = mSymbolFormat;
+            break;
+        default:
```

Comment:
hmm.. I have the feeling that it's better to remove the "default". then we'll get compiler warnings if an enum value is missing.

Reply:
Would this not raise another warning about a missing default in switch statement?

Reply:
Yes.. but I rather suppress that warning and get warnings when I forget enum constants instead.